app/queries/get_unsuccessful_and_unsubmitted_candidates.rb

```diff
@@ -1,41 +1,29 @@
 class GetUnsuccessfulAndUnsubmittedCandidates
   def self.call
     Candidate
-    .joins(:application_forms)
-    .where(
-      application_forms: {
-        recruitment_cycle_year: RecruitmentCycle.previous_year,
-        id: ApplicationChoice.where(status: ApplicationStateChange::UNSUCCESSFUL_STATES).select(:application_form_id),
-      },
-    )
-    .where.not(
-      application_forms:
-      {
-        id: ApplicationChoice.where(status: ApplicationStateChange::SUCCESSFUL_STATES)
-        .select(:application_form_id),
-      },
-    )
-    .where.not(unsubscribed_from_emails: true)
-    .or(
-      Candidate.joins(:application_forms)
-      .where(application_forms:
-        {
-          submitted_at: nil,
-          recruitment_cycle_year: RecruitmentCycle.previous_year,
-        })
-      .where.not(unsubscribed_from_emails: true),
-    )
-    .or(
-      Candidate
-      .joins(:application_forms)
-      .where(application_forms:
-        {
-          recruitment_cycle_year:
-          RecruitmentCycle.current_year,
-        })
-      .where.not(unsubscribed_from_emails: true),
-    )
-    .includes(:application_choices)
-    .distinct
+      .left_outer_joins(:application_forms)
+      .left_outer_joins(application_forms: :application_choices)
+      .where(application_forms: { recruitment_cycle_year: 2023 })
+      .where(application_choices: {
+        status: %w[withdrawn cancelled rejected declined conditions_not_met offer_withdrawn application_not_sent unsubmitted],
+      })
```

Comment:
It looks like this condition will find any candidate that has at least 1 unsuccessful application choices, regardless of whether they have other successful application choices. So I think this is going to return too many records. e.g. in my local DB I got a candidate who had 2 `rejected` and one `offer` application choices.

When we've had to do this kind of query in the past I think we've sometimes used a `NOT EXISTS` condition but I don't know whether the performance would be adequate.

Reply:
thanks you Steve 🙏 

I think we should add a 

```
.where.not(application_choices: {
  status: %w[recruited pending_conditions offer offer_deferred]
})
```

Reply:
This is done! Thank you @stevehook & @vassyz